python/ray/data/datasource/bigquery_datasink.py

```diff
@@ -46,16 +48,22 @@ def on_write_start(self) -> None:
         dataset_id = self.dataset.split(".", 1)[0]
         try:
             client.get_dataset(dataset_id)
-            logger.info(
-                f"Dataset {dataset_id} already exists. "
-                "The table will be overwritten if it already exists."
-            )
         except exceptions.NotFound:
             client.create_dataset(f"{self.project_id}.{dataset_id}", timeout=30)
             logger.info("Created dataset " + dataset_id)
 
-        # Delete table if it already exists
-        client.delete_table(f"{self.project_id}.{self.dataset}", not_found_ok=True)
+        # Delete table if overwrite_table is True
+        if self.overwrite_table:
+            print(
```

Comment:
Any reason we'd want to use `print` instead of `logger` here?
```suggestion
            logger.info(
```

Also, should this be a `logger.debug`? (Not sure if `info` or `debug` is more appropriate here)